Automattic/wp-calypso
# Giphy Block

#### Changes proposed in this Pull Request

The Giphy block is a straightforward integration that allows the user to explicitly specify a Gif by URL or search by keyword. The keyword search is likely to be the most common use. 

Note: the block resizing is based on the approach described in this article https://blog.theodo.fr/2018/01/responsive-iframes-css-trick/

#### Testing instructions

Jurassic Ninja
- https://jurassic.ninja/create?gutenpack&calypsobranch=try/giphy&branch=try/giphy
- Connect Jetpack
- Navigate to Settings->Jetpack Constants, check `JETPACK_BETA_BLOCKS`, and Save.

1) Add the block
2) Go to Giphy, find any Gif you like, copy the URL and paste into the block. The gif should show up.
3) Paste in a Giphy URL following this syntax: http://i.giphy.com/4ZFekt94LMhNK.gif. Past the URL into the block. The gif should show up.
4) Type any text in to the search field. DIfferent Gifs should show up for anything you search for. 
5) Verify that the text field appears when you click on the block. 
6) Verify that the text field disappears after 3.5 seconds of inactivity, unless no search text has been provided in which case it remains visible.
7) Verify caption behaves much like the core Image block.
8) Verify block alignment options all work as expected.
